src/main/java/uk/gov/companieshouse/web/accounts/util/Navigator.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.companieshouse.web.accounts.util;
+
+import java.lang.annotation.Annotation;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.view.UrlBasedViewResolver;
+import org.springframework.web.util.UriTemplate;
+import uk.gov.companieshouse.web.accounts.annotation.NextController;
+import uk.gov.companieshouse.web.accounts.annotation.PreviousController;
+import uk.gov.companieshouse.web.accounts.exception.MissingAnnotationException;
+
+public class Navigator {
+
+    // Private constructor to prevent instantiation
+    private Navigator() {}
+
+    public static String getNextControllerRedirect(Class clazz, String... pathVars) throws MissingAnnotationException {
+        Annotation nextControllerAnnotation = AnnotationUtils.findAnnotation(clazz, NextController.class);
+        if (nextControllerAnnotation == null) {
+            throw new MissingAnnotationException("Missing @NextController annotation on class " + clazz.toString());
+        }
+
+        Class nextControllerClass = ((NextController) nextControllerAnnotation).value();
```

Comment:
worth checking if the class object is null like you're doing for previous?

Reply:
Good point, addressed below